packages/nextjs/app/builders/0x981ab0D817710d8FFFC5693383C00D985A3BDa38/page.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
```

Comment:
Your bio page doesn't seem to have reactive elements, so if you remove this "use client" directive you'll get [a few nice benefits](https://nextjs.org/docs/app/building-your-application/rendering/server-components#benefits-of-server-rendering) for the page.

Reply:
page has 2 `useState` for ENS resolution, these are reactive ?

without "use client" got this error : 
```
./app/builders/0x981ab0D817710d8FFFC5693383C00D985A3BDa38/page.tsx
ReactServerComponentsError:

You're importing a component that needs useState. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.
Learn more: https://nextjs.org/docs/getting-started/react-essentials
```

Reply:
This is perfect chance for you to practice [integrating client-side and server-side elements in the same page](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming), a common pattern in NextJS 14. There are two approaches explained on that page, loading.js and Suspense boundaries.

@carletex or @technophile-04, what pattern would you recommend so zapaz can get the benefits of SSR? I was going to suggest he extract the state and BlockieAvatar to a separate component, then import and wrap in a Suspense boundary. Something like this.
```suggestion
          <Suspense fallback={<span className="loading loading-spinner loading-lg"></span>}>
            <BlockieAvatar address={ensAddress || ""} ensImage={ensAvatar} size={100} />
          </Suspense>
```

Reply:
in order to publish dapp on decentralized storage (like IPFS), I always prefer to use client side only :-)

but happy to learn how to mix client and server side in react!

Reply:
> I was going to suggest he extract the state and BlockieAvatar to a separate component,

Yeah, this is a nice approach, also we don't need suspense. 

We can create a `ZapazAvatar` component (this will be client component) and contains state/ hooks. 
![Screenshot 2024-05-16 at 1 26 40 PM](https://github.com/BuidlGuidl/batch5.buidlguidl.com/assets/80153681/4ec635fd-71bd-4288-bc83-0859e6e039bd)

<details>
<summary>
ZapazAvatar Component : 
</summary>

```tsx
"use client";

import { useEnsAvatar } from "wagmi";
import { BlockieAvatar } from "~~/components/scaffold-eth";

const ensName = "zapaz.eth";
export const ZapazAvatar = () => {
  const ensAddress = "0x981ab0D817710d8FFFC5693383C00D985A3BDa38";
  const { data: ensAvatar, error: errorAvatar } = useEnsAvatar({ name: ensName, chainId: 1 });
  if (errorAvatar) console.error("ENS Avator Error:", errorAvatar);

  return (
    <>
      {ensAvatar ? (
        <BlockieAvatar address={ensAddress} ensImage={ensAvatar} size={100} />
      ) : (
        <span className="loading loading-spinner loading-lg"></span>
      )}
    </>
  );
};

```

</details>

And remove "use client" from `page.tsx`, So that we don't mark whole page as Client component and get benefits of server components in page. 

Checkout [this](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#interleaving-server-and-client-components) for more details, but lol its big rabbit hole of server and client components and hard to wrap your head around. 

> in order to publish dapp on decentralized storage (like IPFS), I always prefer to use client side only :-)

Lol yup, if you are deploying dapp to IPFS then it doesn't add too much...But I think its nice to utilize server components incase you plan to deploy it to vercel  also I think Fleek is also building infra so that they support server components for next apps on IPFS

Reply:
I am using https://fleek.co for quite a long time and now the new https://fleek.xyz ... but never used server side for web3 in order to publish to IPFS :-(

not sure server side is possible on fleek (https://www.reddit.com/r/ipfs/comments/v1c9a8/serverside_and_clientside_rendering_using_fleek/)   

and fyi, we have just published today, via fleek, the latest version of our dapp => https://app.kredeum.com 

now back to react...